src/main/java/io/vertx/core/buffer/Buffer.java

```diff
@@ -36,7 +37,7 @@
  * @author <a href="http://tfox.org">Tim Fox</a>
  */
 @VertxGen
-public interface Buffer extends ClusterSerializable {
+public interface Buffer extends ClusterSerializable, Shareable {
```

Comment:
`Shareable` could also be applied to `JsonObject` and `JsonArray`. Then the code in `Checker` could be further simplified.

Reply:
I had not noticed that comment. But I see no reason why we'd do it for `Buffer` but not `JsonObject` and `JsonArray`.

Reply:
actually I did not see that copyIfRequired was copying them, I thought only buffer was copied